lib/src/pipeline.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'client.dart';
+import 'middleware.dart';
+
+/// A helper that makes it easy to compose a set of [Middleware] and a
+/// [Client].
+///
+///     var client = const Pipeline()
+///         .addMiddleware(loggingMiddleware)
+///         .addMiddleware(basicAuthMiddleware)
+///         .addHandler(new Client());
```

Comment:
`addClient()`

It may make sense to *also* add an `addHandler()` method, though, which automatically creates a `Client`.